### PR TITLE
Steal enhanced scala config from rawls

### PR DESCRIPTION
**Pull request by helgridly**
No version bump necessary! Yessssss.

**PR checklist**
- [x] For each library you've modified here, decide whether it requires a major, minor, or no version bump. (Click [here](/broadinstitute/workbench-libs/blob/develop/CONTRIBUTING.md) for guidance)

If you're doing a **major** or **minor** version bump to any libraries:
- [ ] Bump the version in `project/Settings.scala` `createVersion()`
- [ ] Update `CHANGELOG.md` for those libraries
- [ ] I promise I used `@deprecated` instead of deleting code where possible

In all cases:
- [x] Replace the appropriate version hashes in `README.md` and the `CHANGELOG.md` for any libs you changed with `TRAVIS-REPLACE-ME` to auto-update the version string
- [ ] Get two thumbsworth of PR review
- [ ] Verify all tests go green (CI _and_ coverage tests)
- [ ] Squash commits and **merge to develop**
- [ ] Delete branch after merge

### CODE REVIEW

**Review by MatthewBemis — COMMENTED** [review]
`TRAVIS-REPLACE-ME` for the hash?

**Comment by helgridly** [comment]
travis

### replace

# YOU

**Review by MatthewBemis — APPROVED** [review]
this might be the nicest code i've ever seen :+1:

**Comment by helgridly** [comment]
doug wrote it originally but thanks

File: util/src/main/scala/org/broadinstitute/dsde/workbench/util/EnhancedScalaConfig.scala [hunk 1]
```diff
@@ -0,0 +1,32 @@
+package org.broadinstitute.dsde.workbench.util
+
+import com.typesafe.config.{Config, ConfigException, ConfigObject}
+
+import scala.util.{Failure, Success, Try}
+
+object ScalaConfig {
```
Comment (Qi77Qi) [inline util/src/main/scala/org/broadinstitute/dsde/workbench/util/EnhancedScalaConfig.scala:7]: Personally I don't think this is needed..There're way better config libraries taking care of these things. I don't think this is a custom pattern we should encourage.

**Pull request merged**